packages/yourloops/components/dashboard-widgets/device-usage-widget.tsx

```diff
@@ -93,7 +93,7 @@ export const DeviceUsageWidget: FC<DeviceUsageWidgetProps> = (props) => {
   const { t } = useTranslation()
   const { classes } = useStyles()
   const trackMetric = metrics.send
-  const pumpSettings = [...medicalDataService.grouped.pumpSettings].pop() as PumpSettings
+  const pumpSettings = medicalDataService.medicalData.pumpSettings.slice(-1)[0] as PumpSettings
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Just an idea because this line of code makes me think of it: if we know that we should always have one element in the `medicalData.pumpSettings` array, could the API send us only this element instead of an array? It would allow us to avoid the computation of the last pump settings every time we need it

Reply:
Agree, I created a ticket to work on this later (next release)
https://diabeloop.atlassian.net/browse/YLP-2695